### ISSUE TITLE
How to render dust template located in folders?

### ISSUE THREAD

**Issue by mickyloranger** [linkedin/dustjs#707]
Hi,

I have the following set up:

app.engine('dust', engines.dust);
app.set('view engine', 'dust');
app.set('views', path.join(__dirname,  path.join('public', 'templates')));

Now in /public/templates/ i have many subfolders. For expl sub1, sub2 where my templates are located. How can i from  the folder /routes/sub1/ render the templates located in /public/templates/sub1.

I have try this:
res.render('sub1/index', {title: 'Sub1'});
res.render('sub1.index', {title: 'Sub1'});

but this don't work

Regards
Micky

**Comment by sethkinast** [comment]
What is `engines`? Are you using consolidate? If not, which rendering engine?

What express version?

**Comment by mickyloranger** [comment]
yes i'm using consolidate

**Comment by sethkinast** [comment]
This looks pretty reasonable. I will be able to test an example tomorrow. Until then:

Consolidate expects your files to end in .dust unless you pass it a different extension.

If you aren't attached to consolidate, we have an example in the repo using Hoffman: https://github.com/linkedin/dustjs/blob/master/examples/streaming-hoffman/app.js

**Comment by mickyloranger** [comment]
I change consolidate with hoffman and i can render with res.render('sub1/index', {title: 'Sub1'}); But the engine still look at the layout.dust (i'm using partial) in the /public/template folder instead of /public/template/sub1/

**Comment by sethkinast** [comment]
Yes, template paths in dust are absolute, not relative. I believe a couple
people have worked on renderers that automatically transform relative paths.

In dust, template names have to be unique, so you can't register two
templates under the name "layout", even if they are in separate folders
(the file path is just an implementation detail).

**Comment by mickyloranger** [comment]
can you please me give more infos about about people who are worked on that?
